### DISCUSSION TITLE
Documentation needed for new Client(streambuilder, options);

### DISCUSSION THREAD

**Discussion by FreTimmerman** [mqttjs/MQTT.js#1827] [Q&A]
There seems to be no documentation whatsoever about usage of `mqtt.Client(streamBuilder, options)`
specifically what this `streamBuilder` is or does. but also the instancing of `new Client`

there is mentioned in the notes for v5:
> When creating `MqttClient` instance `new` is now required.

but that's it. no mention of how to use this.

**Comment by robertsLando** [comment]
Like the name saids that's just a function that takes the client and the options as arguments and returns a stream. The most basic one is: https://github.com/mqttjs/MQTT.js/blob/9a18610daf7b350476c1d92347dc24f56882887b/src/lib/connect/tcp.ts#L12

Reason why there is no docs about it is that the `connect` function actually performs some validations on options and other things that you may miss when creating the client with just `new Client`. So generally no one should use it. If you would like to get the client instance you can use the option `manualConnect` so mqtt.connect simply returns the client instance and you can call client.connect later
